File: 2021_spring_winter_algorithm_study_basic/WEEK04_Stack_Queue_Dequeue/1158.py
"""
input :
7 3

output :
<3, 6, 2, 7, 5, 1, 4>
"""

import sys
from collections import deque

sys.stdin = open("input.txt")
input = sys.stdin.readline

N, K = map(int, input().split())

nums = deque([i for i in range(1, N + 1)])

ans = "<"
while True:
    nums.rotate(-K + 1)
    if len(nums) == 0:
        break
    ans += str(nums.popleft()) + ", "
ans = ans[:-2] + ">"

print(ans)

File: 2021_summer_algorithm_study_basic/using_cpp/1158.cpp
/*
input :
7 3

output :
<3, 6, 2, 7, 5, 1, 4>
*/

#include <iostream>
#include <deque>
#include <algorithm>
using namespace std;

int N, K;
int f;
int buf;
deque<int> nums;
int main(){
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    freopen("input.txt", "r", stdin);

    cin >> N >> K;
    for (int i = 1; i <= N; i++){
        nums.push_back(i);
    }

    cout << '<';
    while (true){
        for (int i = 0; i < K % nums.size(); i++){
            f = nums.front();
            nums.pop_front();
            nums.push_back(f);
        }
        buf = nums.back();
        nums.pop_back();

        if (!nums.empty()){
            cout << buf << ", ";
        }
        else{
            cout << buf;
            break;
        }
    }
    cout << '>';

    return 0;
}